### PR TITLE
Add tables for `kolide_falconctl_systags` and `kolide_falconctl_options`

### PR DESCRIPTION
**Pull request by directionless**
Add tables to linux for parsing `falconctl` options.

Experimentally, `systags` needs to be its own thing. But it looks like it's simple enough we can use a fairly straight forward parser for it. 

The options are a bit tricker. The format feels under constrained, and it's not clear how we can work with these. We've found if there's an issue with one, then the entire command line line fails to provide output. To handle this, the table supports setting which options you'd like. Either one exec, or many execs. This option handling proved a bit tricker for the existing exec table generators, so it's its own table. 

Fixes: #788 
Closes: #894

### CODE REVIEW

File: pkg/osquery/tables/execparsers/simple_array/parser.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+type parser struct {
+}
+
+func (p parser) Parse(reader io.Reader) (any, error) {
+	return parse(reader)
+}
+
+func New() parser {
```
Comment (directionless) [inline pkg/osquery/tables/execparsers/simple_array/parser.go:17]: This should take a `key` and add the values as an array

File: pkg/osquery/tables/crowdstrike/falconctl/table.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		"--rfm-reason",
+		"--rfm-state",
+		"--tags",
+		"--version",
+	}
+
+	defaultOption = strings.Join(allowedOptions, " ")
+)
+
+type execFunc func(context.Context, log.Logger, int, []string, []string) ([]byte, error)
```
Comment (James-Pickett) [inline pkg/osquery/tables/crowdstrike/falconctl/table.go:38]: really love this pattern!

File: pkg/osquery/tables/crowdstrike/falconctl/table.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+			if !optionAllowed(option) {
+				level.Info(t.logger).Log("msg", "requested option not allowed", "option", option)
+				// Consider this a fatal error, don't just go on.
+				return nil, fmt.Errorf("requested option not allowed: %s", option)
+			}
+		}
+
+		rowData := map[string]string{"options": requested}
+
+		args := append([]string{"-g"}, options...)
```
Comment (James-Pickett) [inline pkg/osquery/tables/crowdstrike/falconctl/table.go:85]: what does this do? can you leave a comment please?

**Review by James-Pickett — APPROVED** [review]
Looks good, like the use of the noop exec func for testing! I have not tested this due to lack of linux + crowdstrike.

File: pkg/osquery/tables/crowdstrike/falconctl/table.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+var (
+	falconctlPaths = []string{"/opt/CrowdStrike/falconctl"}
+
+	// allowedOptions is the list of options this table is allowed to query. Notable exceotions
```
[the remark this answers is not in the document]
Comment (directionless) [reply pkg/osquery/tables/crowdstrike/falconctl/table.go:20]: tsk. mispell should have caught this. Thanks!

**Comment by directionless** [comment]
> Looks good, like the use of the noop exec func for testing! I have not tested this due to lack of linux + crowdstrike.

Me neither. I'm hoping @FritzX6 can test this.

File: pkg/osquery/tables/crowdstrike/falconctl/table.go (hunk 3 above)
Comment (directionless) [reply pkg/osquery/tables/crowdstrike/falconctl/table.go:85]:
```suggestion
		// As I understand it the falconctl command line uses `-g` to indicate it's fetching the options settings, and
		// then the list of options to fetch. Set the command line thusly. 
		args := append([]string{"-g"}, options...)
```

File: pkg/osquery/tables/crowdstrike/falconctl/table.go [hunk 5]
```diff
@@ -82,6 +82,8 @@ func (t *falconctlOptionsTable) generate(ctx context.Context, queryContext table
 
 		rowData := map[string]string{"options": requested}
 
+		// As I understand it the falconctl command line uses `-g` to indicate it's fetching the options settings, and
+		// then the list of options to fetch. Set the command line thusly.
```
Comment (James-Pickett) [inline pkg/osquery/tables/crowdstrike/falconctl/table.go:86]: Thusly .... I did not know that was a word ... checks out on google 👍

**Review by James-Pickett — DISMISSED** [review]

**Review by seejdev — APPROVED** [review]

**Pull request merged**